cps/main.c

```diff
@@ -641,14 +644,25 @@ kni_create(struct rte_kni **kni, struct rte_mempool *mp,
 		conf.group_id = rte_eth_bond_primary_get(iface->id);
 	else
 		conf.group_id = iface->id;
+
+	memset(&dev_info, 0, sizeof(dev_info));
 	rte_eth_dev_info_get(conf.group_id, &dev_info);
-	conf.addr = dev_info.pci_dev->addr;
-	conf.id = dev_info.pci_dev->id;
+	if (dev_info.device)
+		bus = rte_bus_find_by_device(dev_info.device);
+	if (bus && !strcmp(bus->name, "pci")) {
```

Comment:
This whole `if` statement should be under the `if` statement `if (dev_info.device)`, this way, variable `bus` can be local to the `if` statement `if (dev_info.device)`.